crates/sui/src/client_commands.rs

```diff
@@ -2167,3 +2213,45 @@ impl Display for SwitchResponse {
         write!(f, "{}", writer)
     }
 }
+
+/// Request tokens from the Faucet for the given address
+pub async fn request_tokens_from_faucet(
+    address: SuiAddress,
+    url: String,
+) -> Result<(), anyhow::Error> {
+    let address_str = address.to_string();
+    let json_body = json![{
+        "FixedAmountRequest": {
+            "recipient": &address_str
+        }
+    }];
+
+    // make the request to the faucet JSON RPC API for coin
+    let client = reqwest::Client::new();
+    let resp = client
+        .post(&url)
+        .header("Content-Type", "application/json")
+        .json(&json_body)
+        .send()
+        .await?;
+    println!("Requested gas from faucet: {url}");
+    println!(
+        "Faucet request for address {address_str} has status: {}",
+        resp.status()
+    );
+    if resp.status() == 429 {
+        bail!("Faucet received too many requests from this ip address. Please try again after 60 minutes.");
+    }
+    println!("Waiting for the faucet to complete the gas request...");
+
+    let faucet_resp: FaucetResponse = resp.json().await?;
+
+    let task_id = if let Some(err) = faucet_resp.error {
+        bail!("Faucet request was unsuccessful. Error is {err:?}")
+    } else {
+        faucet_resp.task
+    };
+    println!("Faucet task id: {task_id}");
+    println!("The coin should be available in the next couple of minutes. Run sui client gas to find your coins.");
```

Comment:
It is usually quicker than a couple of minutes, so is it worth waiting for it here? (Perhaps guarded by a flag/option?)

Reply:
This is something that we can follow up with as well!

Reply:
The issue with waiting is that often due to the rate limiter, we cannot make a request to the faucet server for the status as it will return 429. The hacky way is to call `gas` command, and then in a loop call the gas command and get the coins until we have coins.len() + 1. 
Maybe a `--wait-for-coin-id` flag can be used to indicate that the user wants to wait to get the coin id? 

The main issue is that I could have my set active network to X, and then issue `faucet --url Y`, so when calling `gas` command, it will use the X network instead of the Y one...

Reply:
Tried to do something and realized that if a custom RPC's faucet endpoint does not follow the usual `faucet.network.address` it's hard to "deconstruct" the RPC endpoint, so it is not so straightforward. For now, it will just tell the user to check for gas. If anyone has better ideas, I am all ears :D